semgrep_output_v1.atd

```diff
@@ -1254,10 +1256,19 @@ type historical_configuration = {
     ?lookback_days: int option;
 }
 
+type glob = string
+
+type product_ignored_files = (product * glob list) list
```

Comment:
@mjambon what are the consequences if tomorrow we add a new product, so the product enum gets a new entry.
What will do old versions of semgrep if suddenly they get in those lists a new product? The element is skipped?
An error is generated? What about those association lists?

Reply:
We could implement an error-tolerance mechanism similar to what tree-sitter does: parsing errors bubble up until reaching a node that can be skipped safely i.e. a list element, optional, or nullable. In the case of an invalid `product` in `(product * glob list) list`, the invalid JSON pair `["New_product_kind", [...]]` would be removed from the list and kept aside as an error. This work would have to be done for each programming language that wants this feature (atdpy for Python, atdgen for OCaml) except that I don't want to even try to add such features to atdgen because it's too complicated. It would be reasonable to do in atdpy, atdts, or a [future atdml reimplementation of atdgen](https://github.com/ahrefs/atd/issues/313) that simplifies things.

Reply:
Will atdcheck report those regressions? I mean if I add a new enum in Product, will atdcheck complains that suddenly the product_ignored_files becomes backward incompatible?

Reply:
So in the end, using strings is actually more robust to evolution atd-wise; but anyway it would just delay the issue
because we would need to convert those strings in actual Product type later